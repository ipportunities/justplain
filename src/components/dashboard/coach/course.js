import React, { useState, useEffect, useRef } from "react";
import { useSelector, useDispatch } from "react-redux";
import { setIntervention } from "../../../actions";
import { useHistory, useParams } from "react-router-dom";
import apiCall from "../../api";
import t from "../../translate";
import LeftMenu from "../leftmenu";
import AgendaChatcourse from "../../agenda/chatcourse";
import Texts from "../../texts";
import Groups from "../../groups";
import GroupsArchive from "../../groups/archive";
import Registrations from "../../registrations";
import Registration from "../../registrations/single";
import Students from "../../students";
import DashboardCoachNavbar from "./navbar";
import { useLocation } from "react-router-dom";
import $ from "jquery";

const CourseDashboard = props => {

  let location = useLocation();
  const dispatch = useDispatch();
  const history = useHistory();
  const { interventionId } = useParams()

  const auth = useSelector(state => state.auth);
  const intervention = useSelector(state => state.intervention);
  const uiTranslation = useSelector(state => state.uiTranslation);

  const [interventions, setInterventions] = useState([])
  const [registrationID, setRegistrationID] = useState(0)

  const settingsType = location.pathname.split('/')[3]
  const [activeTab, setActiveTab] = useState('')

  ////Ten behoeve van het klikken in het hoofdmenu opdat detail views gereset kunnen worden
  ///TODO nog een keer nagaan of dit gegroeide wel zo logisch is...
  const [studentId, setStudentId] = useState(0)
  const [studentTab, setStudentTab] = useState('')
  const [activeGroup, setActiveGroup] = useState('')

  if (settingsType !== activeTab && typeof settingsType != 'undefined' && settingsType !== '' && !Number.isInteger(parseInt(settingsType))) {
    setActiveTab(settingsType)
  }

  const showTab = tab => {
    setActiveTab('');
    if(tab == 'groups' || tab == 'group-archive'){
      setActiveGroup('');
    }
    //setActiveTab(tab);
  };

  useEffect(() => {
    //api aanroepen om settings op te halen
    getInterventionSettings(interventionId)
  }, [auth]);

  useEffect(() => {
    setActiveTab('');
    if(location.pathname.split("/")[3] != activeTab) {

      if (typeof location.pathname.split("/")[3] !== "undefined") {
        setActiveTab(location.pathname.split("/")[3]);
      } else {
        setActiveTab('registrations')
      }
    }
    if (location.pathname.split("/")[3] === 'registrations') {
      if (typeof location.pathname.split("/")[4] !== 'undefined' && Number.isInteger(parseInt(location.pathname.split("/")[4])) && parseInt(location.pathname.split("/")[4]) > 0) {
        setRegistrationID(parseInt(location.pathname.split("/")[4]))
      } else {
        setRegistrationID(0)
      }
    }
    if(location.pathname.split("/")[3] == "students" && location.pathname.split("/").length >= 5){
      setStudentId(location.pathname.split("/")[4])
      if(location.pathname.split("/").length > 5){
        setStudentTab(location.pathname.split("/")[5])
      }
    } else {
      setStudentId(0)
    }
  }, [location]);

  let scrollbarRef = useRef()

  const handleScroll = (event) => {
    let scrollTop = $(document).scrollTop();
    if(scrollTop > 180){
      $(".coachInterface").addClass("navScrolledTop")
    } else {
      $(".coachInterface").removeClass("navScrolledTop")
    }
  }

  //onFirstRenderOnly to prevent infinite render loop
  useEffect(() => {

    if(props.interventions != ""){
      setInterventions(props.interventions)
    }

    window.addEventListener('scroll', handleScroll);

  }, [props]);

  function changeIntervention(int_id){

    let this_intervention_obj = interventions.filter(function (intervention) {
      return intervention.id === int_id
    });
    if(this_intervention_obj.length != 0){

      history.push("/dashboard/" + int_id)
      getInterventionSettings(int_id)
    }

  }

  const getInterventionSettings = (intervention_id) => {
    apiCall({
      action: "get_intervention_settings",
      token: auth.token,
      data: {
        intervention_id,
        include_forms: true, //tijdelijk...
        language_id: uiTranslation.language_id

      }
    }).then(resp => {
      //set settings in global state:
      dispatch(
        setIntervention(
          intervention_id,
          resp.organisation_id,
          resp.title,
          resp.settings
        )
      );
    });

  }

  return (
    <div className="coachInterface coachInterface_v2">
      <table className="navbar navbar-coachInterface_v2 navbar-expand-lg navbar-light bg-light">
        <tbody>
          <tr>
            <td>
              <h2 className='select'>
                <select value={intervention.id} onChange={(e) => changeIntervention(e.target.value)}>
                  {interventions.map((int, index) => (
                    <option value={int.id} key={index}>
                      {int.title}
                    </option>
                  ))}
                </select>
                {intervention.title}
              </h2>
            </td>
            <td>
            <DashboardCoachNavbar showTab={showTab}/>
            </td>
          </tr>
        </tbody>
      </table>
      <LeftMenu />
      <div className="content">
        {activeTab == "students" ? <Students interventions={interventions} hasMenu={true} studentId={studentId} studentTab={studentTab} setStudentId={setStudentId} />:<></>}
        {activeTab == "agenda" ? <AgendaChatcourse />:<></>}
        {activeTab == "texts" ? <Texts />:<></>}
        {activeTab == "groups" ? <Groups activeGroup={activeGroup} setActiveGroup={setActiveGroup}/>:<></>}
        {activeTab == "group-archive" ? <GroupsArchive activeGroup={activeGroup} setActiveGroup={setActiveGroup}/>:<></>}
        {
          activeTab == "registrations" && registrationID !== 0 ? <Registration registration_id={registrationID} /> :
            activeTab == "registrations" ? <Registrations /> : <div>{activeTab} {registrationID}</div>
        }
      </div>
    </div>
  )
}
export default CourseDashboard
