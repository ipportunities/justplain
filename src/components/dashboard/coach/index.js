import React, {useEffect, useState} from "react";
import { useSelector } from "react-redux";
import { useHistory } from "react-router-dom";
import { Redirect } from "react-router";
import InterventionLinks from "../../interventionlinks";
import t from "../../translate";
import LeftMenu from "../leftmenu";

const Coach = props => {

  let history = useHistory();

  const auth = useSelector(state => state.auth);

  const [loading, setLoading] = useState(true);

  const onClickHandler = (event, intervention) => {
    if (typeof intervention.settings !== "undefined" && typeof intervention.settings.intervention_type !== "undefined" && intervention.settings.intervention_type === "chatcourse") {
      history.push("/dashboard/" + intervention.id);
    } else {
      history.push("/students/" + intervention.id);
    }
  }

  useEffect(() => {
    if (typeof props.interventions !== "undefined" && props.interventions.length > 0 && typeof auth.rights.interventions !== "undefined") {
      setLoading(false)
    }
  }, [props.interventions, auth.rights.interventions])

  if (!loading) {
    if (auth.rights.interventions.length < 1) {
      return (
        <div className="transWrapper">
          <div className="container dashboard_container">
            {t("Er zijn momenteel geen interventies aan uw account gekoppeld.")}
          </div>
        </div>
      );
    } else {
      if (auth.rights.interventions.length == 1 || props.interventions.length == 1) {
        let intervention = false;
        auth.rights.interventions.forEach(interv1 => {
          props.interventions.forEach(interv2 => {
            if (parseInt(interv1.id) === parseInt(interv2.id)) {
              intervention = interv2;
            }
          });
        });

        if (intervention.settings.intervention_type ===  'chatcourse') {
          return <Redirect to={"/dashboard/" + intervention.id} />;
        } else {
          return <Redirect to={"/students/" + intervention.id} />;
        }

      } else {
        return (
          <div className="transWrapper">
            <div className="container dashboard admin">
              <LeftMenu />
              <InterventionLinks
                refreshState="0"
                onClickHandler={onClickHandler}
                interventions={props.interventions}
              />
            </div>
          </div>
        );
      }
    }
  } else {
    return (
      <div className="transWrapper">
          <div className="container dashboard_container">
            {t("Loading...")}
          </div>
        </div>
    )
  }
}

export default Coach;
