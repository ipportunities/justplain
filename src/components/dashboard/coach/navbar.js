import React, { useEffect, useState } from "react";
import { useHistory } from "react-router-dom";
import { useSelector } from "react-redux";
import t from "../../translate";
import { useLocation } from "react-router-dom";
import Saved from "../../intervention/saved";
import {appSettings} from "../../../custom/settings";

const DashboardCoachNavbar = props => {
  let location = useLocation();
  const intervention = useSelector(state => state.intervention);
  const history = useHistory();

  const [navtab, setNavtab] = useState("");
  const settingsType = location.pathname.split("/")[3];

  if(settingsType != navtab && typeof settingsType != "undefined"  && settingsType != "")
  {
    setNavtab(settingsType);
  }

  useEffect(() => {
    if(location.pathname.split("/")[3] != navtab){
      setNavtab(location.pathname.split("/")[3]);
    }
  }, [location]);

  const navigateTo = gotab => {
    history.push("/dashboard/" + intervention.id + "/" + gotab + "/");
    setNavtab(gotab);
    props.showTab(gotab);
    window.scrollTo(0, 0);
  };

  return (
    <nav className="">
      <Saved />
      <div id="navbarNavAltMarkup">
        <div className="navbar-nav">
          {intervention.settings.intervention_type == "chatcourse" ?
            <>
              <span
                id="settings-navbar-registrations"
                className={
                  navtab == "registrations"
                    ? "nav-item nav-link active"
                    : "nav-item nav-link"
                }
                onClick={() => {
                  navigateTo("registrations");
                }}>
              {t("Aanmeldingen")}
              </span>
            </>
              :
              <></>
            }
            <span
              id="settings-navbar-students"
              className={
                navtab == "students"
                  ? "nav-item nav-link active"
                  : "nav-item nav-link"
              }
              onClick={() => {
                navigateTo("students");
              }}
            >
              {t("Deelnemers")}
            </span>
            {intervention.settings.intervention_type == "chatcourse" ?
              <>
              <span
                id="settings-navbar-groups"
                className={
                  navtab == "groups"
                    ? "nav-item nav-link active"
                    : "nav-item nav-link"
                }
                onClick={() => {
                  navigateTo("groups");
                }}
              >
                {t("Groepen")}
                </span>
              <span
                id="settings-navbar-group-archive"
                className={
                  navtab == "group-archive"
                    ? "nav-item nav-link active"
                    : "nav-item nav-link"
                }
                onClick={() => {
                  navigateTo("group-archive");
                }}
              >
                {t("Groepen archief")}
                </span>
              <span
                id="settings-navbar-group-texts"
                className={
                  navtab == "texts"
                    ? "nav-item nav-link active"
                    : "nav-item nav-link"
                }
                onClick={() => {
                  navigateTo("texts");
                }}
              >
                {t("Teksten")}
                </span>
              <span
                id="settings-navbar-group-agenda"
                className={
                  navtab == "agenda"
                    ? "nav-item nav-link active"
                    : "nav-item nav-link"
                }
                onClick={() => {
                  navigateTo("agenda");
                }}
              >
                {t("Planning")}
                </span>
              </>
            :''}
        </div>
      </div>
    </nav>
  );
};

export default DashboardCoachNavbar;
