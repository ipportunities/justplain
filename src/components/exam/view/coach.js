import React from 'react'
import { useTranslation } from '../../translate'
import { useHistory, useParams } from 'react-router-dom'
import { useSelector } from 'react-redux'
import apiCall from '../../api'
import { ViewQuestion } from './question'
import moment from 'moment'
import { ExamMomentStatusDisplay } from '../statusDisplay'
import { ParseText } from '../../../helpers/examTextParser'

export const ExamViewCoach = () => {
  const t = useTranslation()
  const history = useHistory()
  const { examMomentId } = useParams()
  const auth = useSelector(state => state.auth)
  const intervention = useSelector(state => state.intervention)

  const [exam, setExam] = React.useState({})

  const setPoints = (questionId, points) => {
    if (exam.moment.status !== 'generated') {
      const newQuestions = exam.questions
      const maxPoints = newQuestions.find(q => q.id === questionId).maxPoints
      if (points > maxPoints) {
        points = maxPoints
      }
      newQuestions.find(q => q.id === questionId).result.points = points
      setExam({ ...exam, questions: newQuestions })
    }
  }

  const setComment = (questionId, comment) => {
    if (exam.moment.status !== 'generated') {
      const newQuestions = exam.questions
      newQuestions.find(q => q.id === questionId).result.comment = comment
      setExam({ ...exam, questions: newQuestions })
    }
  }

  React.useEffect(() => {
    apiCall({
      action: 'get_exam_for_moment',
      token: auth.token,
      data: { examMomentId: examMomentId }
    }).then(resp => {
      setExam(resp.exam)
    })
  }, [examMomentId])

  const mark = () => {
    apiCall({
      action: 'mark_exam',
      token: auth.token,
      data: {
        examMomentId: examMomentId,
        answers: exam.questions.map(q => {
          return {
            questionId: q.id,
            points: q.result.points,
            comment: q.result.comment
          }
        })
      }
    }).then(() => {
      history.push(`/students/${ intervention.id }`, { user: +exam.userId, tab: 'generatableExamTypes' })
    })
  }

  return Object.keys(exam).length > 0 ? <div>
      <nav className="navbar navbar-expand-lg navbar-light overRule">
        <h2>
          <span className="pointer" onClick={ () => history.push(`/`) }>
              { `${ intervention.title } > ` }
          </span>
          <span className="pointer" onClick={ () => history.push(`/students/${ intervention.id }`) }>
              { `${ t('deelnemers') } > ` }
          </span>
          <span className="pointer"
                onClick={ () => history.push(`/students/${ intervention.id }/`, { user: +exam.userId }) }>
            { `${ exam.userFirstname } ${ exam.userLastname } > ` }
          </span>
          <span className="pointer"
                onClick={ () => history.push(`/students/${ intervention.id }/`, {
                  user: +exam.userId,
                  tab: 'generatableExamTypes'
                }) }>
            { t('Exams') }
          </span>
        </h2>
      </nav>
      <div className="row pt-3">
        <h4 className="col-8 offset-2">{ exam.title }</h4>
      </div>
      <div className="row border-bottom pt-4 pb-3">
        { exam.moment && <>
          <div className="col-4 offset-2">
            { `This exam can be made between ${ moment(exam.moment.startDate).format('DD-MM-YYYY HH:mm') } and ${ moment(exam.moment.endDate).format('DD-MM-YYYY HH:mm') }` }
            <br />
            <ExamMomentStatusDisplay status={ exam.moment.status } />
          </div>
          <div className="col-2">
            { exam.questions.reduce((sum, q) => sum + parseInt(q.maxPoints), 0) } points available <br />
            { exam.questions.reduce((sum, q) => sum + parseInt(q.result ? q.result.points : 0), 0) } points achieved
          </div>
          <div className="col-2 text-right">
            { exam.moment.status !== 'generated' ? <button className="btn btn-success" onClick={ mark }>
              { `${ t('Save') } ${ exam.moment.status === 'filled' ? `& ${ t('Mark checked') }` : '' }` }
            </button> : null }
          </div>
        </>
        }
      </div>
      <div className="row border-bottom pt-2 pb-3">
        <div className="col-8 offset-2" dangerouslySetInnerHTML={ { __html: ParseText(exam.introText, exam) } } />
      </div>
      { exam.questions && exam.questions.map(q => <div className="row border-bottom" key={ q.id }>
        <ViewQuestion question={ q } coachView
                      setPoints={ exam.moment.status !== 'generated' ? (points) => setPoints(q.id, points) : false }
                      setComment={ exam.moment.status !== 'generated' ? (comment) => setComment(q.id, comment) : false } />
      </div>) }
      <div className="row border-bottom pt-2 pb-3">
        <div className="col-8 offset-2" dangerouslySetInnerHTML={ { __html: ParseText(exam.finishText, exam) } } />
      </div>
    </div>
    : <div />
}
