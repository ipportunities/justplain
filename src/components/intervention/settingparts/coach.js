import React, {useState} from "react";
import { useSelector, useDispatch } from "react-redux";
import { setIntervention, setSavingStatus } from "../../../actions";
import t from "../../translate";
import AddImage from "../../content/edit/parts/helpers/addImage";
import MediaLibrary from "../../medialibrary";
import { setChosenImage } from "../../../actions";
import { getClone } from "../../utils";

let saveSettingsTimeout = null;

const InterventionSettingsCoach = props => {
  const dispatch = useDispatch();

  const intervention = useSelector(state => state.intervention);
  const url = useSelector(state => state.url);

  const saveCoachSettings = (newSettings) => {

    dispatch(
      setIntervention(
        intervention.id,
        intervention.organisation_id,
        intervention.title,
        newSettings
      )
    );

    props.setErrorMessage("");
    clearTimeout(saveSettingsTimeout);
    dispatch(setSavingStatus("not_saved"));

    saveSettingsTimeout = setTimeout(() => {
      props.saveSettings(
        intervention.id,
        intervention.organisation_id,
        intervention.title,
        newSettings
      );
    }, 1500);
  }

  ////////////////////////////////////////////
  /////////Media library functies om door te geven
  ///////////////////////////////////////////
  const [mediaLibraryVisible, setMediaLibraryVisible] = useState("hide");
  const medialibrary = useSelector(state => state.mediaLibrary);
  //////////////////////
  ///Save if chosen image uit bieb is not empty alleen bij overeenkomstige id
  if(medialibrary.chosen_image != "" && medialibrary.index == "coachPhoto")
  {
    /// empty chosen image status
    dispatch(
      setChosenImage(
        ''
      )
    );
    let interventionC = getClone(intervention);
    if(medialibrary.index == "coachPhoto"){
      interventionC.settings.coachPhoto = url + "/uploads/intervention/" + medialibrary.chosen_image;
    }
    saveCoachSettings(interventionC.settings);
  }

  function showMediaLibrary(index) {
    setMediaLibraryVisible("show");
    document.getElementsByTagName('html')[0].style.overflow = "hidden";

    /// empty store + set index
    dispatch(setChosenImage("", index));
  }

  function deleteImage(){
    let interventionC = getClone(intervention);
    interventionC.settings.coachPhoto = "";
    saveCoachSettings(interventionC.settings);
  }

  return (
    <div id="settings-coach" className="form-group">
      <MediaLibrary
        mediaLibraryVisible={mediaLibraryVisible}
        setMediaLibraryVisible={setMediaLibraryVisible}
      />
      <div className="question logo">
        <h5>{t("Coach achtergrond afbeelding")}</h5>
        <AddImage index="coachPhoto" showMediaLibrary={showMediaLibrary} image={typeof intervention.settings.coachPhoto != "undefined"?intervention.settings.coachPhoto:''} deleteImage={deleteImage}/>
      </div>
    </div>
  );
};

export default InterventionSettingsCoach;
