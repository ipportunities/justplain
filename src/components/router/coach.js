import React, {useState, useEffect} from "react";
import { BrowserRouter as Router, Switch, Route, Redirect } from "react-router-dom";
import NavBar from "../navBar";
import Dashboard from "../dashboard/coach";
import CourseDashboard from "../dashboard/coach/course";
import Students from "../students";
import InterventionCoaches from "../coaches/";
import apiCall from "../api";
import { useSelector } from "react-redux";

const RouterCoach = () => {

  const auth = useSelector(state => state.auth);
  const intervention = useSelector(state => state.intervention);
  const [supervisorFor, setSupervisorFor] = useState([]);
  const [interventions, setInterventions] = useState([])
  const [isActiveInterventionChatCourse, setIsActiveInterventionChatCourse] = useState(false)

  useEffect(() => {
    //api aanroepen
    apiCall({
      action: "get_interventions",
      token: auth.token,
      data: {}
    }).then(resp => {
      setInterventions(resp.interventions)
    });

    let localSupervisorFor = [];
    auth.rights.interventions.forEach(function (interv, index) {
      if (interv.isSupervisor)
      {
        localSupervisorFor.push(interv.id);
      }
    });
    setSupervisorFor(localSupervisorFor);
  }, []);

  useEffect(() => {

  }, [intervention]);

  return (
    <Router basename={process.env.PUBLIC_URL}>
      <Switch>
        <Route path="/" exact >
          <NavBar />
          <Dashboard interventions={interventions}/>
        </Route>
        <Route path="/students/:interventionId">
          <Students interventions={interventions}/>
        </Route>
        {supervisorFor.indexOf(intervention.id) > -1 ?
          <Route path="/intervention/coaches/" >
            <InterventionCoaches interventions={interventions}/>
          </Route>
        : false}}

        <Route path="/dashboard/:interventionId">
          <CourseDashboard interventions={interventions}/>
        </Route>

        <Route path="*" exact>
          <Redirect to="/" />
        </Route>



      </Switch>
    </Router>
  );
};

export default RouterCoach;
