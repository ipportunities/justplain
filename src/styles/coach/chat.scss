.student_details {
  .chat{
      z-index: 0;
      position: relative;
      header{
        .coach{
          font-size: 19px;
          font-weight:bold;
        }
        border-bottom: 2px solid #A1BDCA;
        margin-bottom: 20px;
        padding-bottom: 20px;
      }
      .illustration{
        position: absolute;
        right: -200px;
        z-index: -1;
        margin-top: -125px;
        opacity: 0.5;
      }
      .chatContent{
        z-index: 1;
        background: #fff;
        padding: 20px 0;
        td:nth-child(2){
          vertical-align: top;
        }
        td:nth-child(1){
          padding-right: 20px;
          img{
            width: 64px;
          }
        }
      }
      .newMessage{
        margin-bottom: 20px;
      }
      .messages{
        .message{
          border-radius: 10px;
          padding: 15px;
          color: $dark_bleu;
          width: 85%;
          margin-bottom: 20px;
          .name {
            font-weight: bold;
          }
          .content{
            padding-left: 0px;
            margin-top: 0px;
          }
          .sendingTime{
            width: 100%;
            text-align: right;
            padding-top: 20px;
            color: #aaa;
          }
          &.received{
            background: #FFECD5 0% 0% no-repeat padding-box;
            border-radius: 20px 20px 20px 0;
            opacity: 1;
            .sendingTime{
              /*text-align: left;*/
            }
          }
          &.sent{
            background: #E6EEF2 0% 0% no-repeat padding-box;
            border: 1px solid #00000000;
            border-radius: 20px 20px 0px 20px;
            opacity: 1;
            margin-left: 15%;
          }
          .blue {
            color: #34B7F1;
          }
        }
      }
      .newMessage{
        background: none;
        border:none;
        width: 100%;
        background: #f5f5f5;
        padding: 20px;
        margin:0 0 30px;
        display: block;
      }

      .input_no_bg{
        border:none;
        background: none;
        padding: 0 0;
        max-width: 100%;
      }
    }
}
