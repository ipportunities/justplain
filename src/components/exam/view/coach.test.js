import React from 'react'
import { shallow } from 'enzyme'
import { ExamViewCoach } from './coach'

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn().mockReturnValue({ auth: { token: 'a' } })
}))

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({
    examMomentId: 0,
  }),
  useHistory: () => ({
    push: jest.fn()
  })
}))

describe('Exam View Coach', () => {
  it('should render correctly', () => {
    const component = shallow(<ExamViewCoach />)
    expect(component).toMatchSnapshot()
  })
})
