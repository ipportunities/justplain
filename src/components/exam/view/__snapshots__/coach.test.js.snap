// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exam View Coach should render correctly 1`] = `<div />`;
